### ISSUE TITLE
Solveset throws TypeError with sqrt in denominator

### ISSUE THREAD

**Issue by JMSS-Unknown** [sympy/sympy#15350]
I was trying to solve
`solveset(diff(sqrt(1/x+x)))`
but it returned
`TypeError: 'NoneType' object is not iterable.`
Reducing the problem, it seems to occur if there is a sqrt in the denominator.
`solveset(1/sqrt(x))`
`TypeError: 'NoneType' object is not iterable`
I tested this with Sympy version 1.3 and 1.4.dev and the issue occured in both versions.

**Comment by fuomag9** [comment]
1/sqrt(x)=0 has no solutions (https://www.wolframalpha.com/input/?i=1%2Fsqrt(x)%3D0)
if you want to differentiate sqrt(1/x+x) you don't need to use sympy.solveset() but sympy.diff()
```
>>> sympy.diff(sqrt(1/x+x))
(1/2 - 1/(2*x**2))/sqrt(x + 1/x)
```

**Comment by smichr** [comment]
The TyperError is a problem, nonetheless, caused because sovleset is not expecting `unrad` to return None.

**Comment by JMSS-Unknown** [comment]
@fuomag9 Yes, 1/sqrt(x) has no solutions, but diff(sqrt(1/x+x))=0 has solutions x=-1 and x=1 (see https://www.wolframalpha.com/input/?i=D(sqrt(1%2Fx%2Bx),x)%3D0), but does not find it due to the sqrt in the denominator raising an exception.

**Comment by JMSS-Unknown** [comment]
@smichr Isn't the issue with `unrad` as in the docstring it states "None is returned if there are no radicals to remove", but there is a radical in the denominator.

Probably a better example to this issue is
```python
solveset((1-x)/sqrt(x))
```
which should return 1

**Comment by oscargus** [comment]
Now this returns `FiniteSet(-1, 1)` so if a test is added, the issue can be closed.

**Comment by oscarbenjamin** [comment]
Fix bisected to d237dcee792da526ee262d6f20211923ed0bca33 from #18324
